### ISSUE TITLE
Support JSON Path functions for JSONB

### ISSUE THREAD

**Issue by matanper** [risingwavelabs/risingwave#11372]
### Is your feature request related to a problem? Please describe.

TLDR; I want a dynamic processing for json columns.

I want to use RW to process events from many different clients, and to standardize the events. 
For example I have a "profile" json and I want to standardize out of it an "age" field.
For client A, he will have "age" field directly inside the json, for client B he will have "birthday" and I'll have to do "now - profile->birthday"

### Describe the solution you'd like

I imagine using [JMESPath](https://docs.rs/jmespath/latest/jmespath/) , and then having a mapping table which I can join against and then do (with some new '-@>' jmespath operator):

```
select profile-@>age_path as age
from events
join mappings on event.client_id = mappings.client_id
```

### Describe alternatives you've considered

Do the mapping somewhere else before reaching RW

**Comment by matanper** [comment]
Thanks! 
Cool option, I'll try to create a udf using python jmsepath

**Comment by fuyufjh** [comment]
The current design follows the PG's JSON functions such as `->` (to extract a `jsonb`) and `->>` (to extract `text`). PG also supports JSON path:  `jsonb @@ jsonpath`, which is not done in RisingWave yet but in the roadmap. Does this look good to you?

References: https://www.postgresql.org/docs/current/functions-json.html

**Comment by matanper** [comment]
I mostly miss arithmetic operations in jsonpath, for now I'm using the UDF option
